Crash on unnamed output {} block

Originally reported in [vscode-terraform](https://github.com/hashicorp/vscode-terraform/issues/717) but probably more relevant here.

tl;dr unnamed `output{}` block causes a panic:

```
panic: runtime error: index out of range [0] with length 0

goroutine 70 [running]:
github.com/hashicorp/hcl-lang/decoder.resolveBlockAddress(0x140096185a0, 0x1400ffb5400, 0x0, 0x100f7c3b0, 0x0, 0x0)
	github.com/hashicorp/hcl-lang@v0.0.0-20210803155453-7c098e4940bc/decoder/reference_targets.go:1233 +0x570
github.com/hashicorp/hcl-lang/decoder.(*Decoder).decodeReferenceTargetsForBody(0x1400ffa7d28, 0x1400ffaa210, 0x0, 0x1400ffac100, 0x0, 0x0, 0x100f7c3b0)
	github.com/hashicorp/hcl-lang@v0.0.0-20210803155453-7c098e4940bc/decoder/reference_targets.go:362 +0x37c
github.com/hashicorp/hcl-lang/decoder.(*Decoder).CollectReferenceTargets(0x1400ffa7d28, 0x0, 0x0, 0x0, 0x0, 0x0)
	github.com/hashicorp/hcl-lang@v0.0.0-20210803155453-7c098e4940bc/decoder/reference_targets.go:321 +0x1c4
github.com/hashicorp/terraform-ls/internal/terraform/module.DecodeReferenceTargets(0x140000a81a0, 0x100c60628, 0x140000a81c0, 0x1400635a0c7, 0x14, 0x2, 0x1400d50e000)
	github.com/hashicorp/terraform-ls/internal/terraform/module/module_ops.go:375 +0x2bc
github.com/hashicorp/terraform-ls/internal/terraform/module.(*moduleLoader).executeModuleOp(0x1400d4baf50, 0x100c6b118, 0x1400d507900, 0x1400635a0c7, 0x14, 0x7, 0x0, 0x1400ffcafc0)
	github.com/hashicorp/terraform-ls/internal/terraform/module/module_loader.go:186 +0x784
github.com/hashicorp/terraform-ls/internal/terraform/module.(*moduleLoader).run.func1(0x100c6b118, 0x1400d507900, 0x1400635a0c7, 0x14, 0x7, 0x0, 0x1400ffcafc0, 0x1400d4baf50)
	github.com/hashicorp/terraform-ls/internal/terraform/module/module_loader.go:101 +0x54
created by github.com/hashicorp/terraform-ls/internal/terraform/module.(*moduleLoader).run
	github.com/hashicorp/terraform-ls/internal/terraform/module/module_loader.go:100 +0x16c
```

--- a/decoder/reference_targets.go
+++ b/decoder/reference_targets.go
@@ -1226,7 +1226,7 @@ func resolveBlockAddress(block *hclsyntax.Block, addr *schema.BlockAddrSchema) (
 		case schema.StaticStep:
 			stepName = step.Name
 		case schema.LabelStep:
-			if uint(len(block.Labels)-1) < step.Index {
+			if len(block.Labels)-1 < int(step.Index) {
 				// label not present
 				return lang.Address{}, false
 			}

--- a/decoder/reference_targets_test.go
+++ b/decoder/reference_targets_test.go
@@ -4784,6 +4784,29 @@ module "different" {
 				},
 			},
 		},
+		{
+			"block with missing label",
+			&schema.BodySchema{
+				Blocks: map[string]*schema.BlockSchema{
+					"output": {
+						Labels: []*schema.LabelSchema{
+							{Name: "name", IsDepKey: true},
+						},
+						Address: &schema.BlockAddrSchema{
+							Steps: []schema.AddrStep{
+								schema.StaticStep{Name: "output"},
+								schema.LabelStep{Index: 0},
+							},
+						},
+						Body: &schema.BodySchema{},
+					},
+				},
+			},
+			`output {
+}
+`,
+			lang.ReferenceTargets{},
+		},
 	}
 
 	for i, tc := range testCases {

